projects/kit/components/input-date/input-date.component.ts

```diff
@@ -272,4 +278,18 @@ export class TuiInputDateComponent
     ): boolean {
         return nullableSame(oldValue, newValue, (a, b) => a.daySame(b));
     }
+
+    @tuiPure
+    private calculateMask(
```

Comment:
Looks like we specifically did not add recalculation of the mask so that if you change min/max it will not update the text (because it will only update text and not the actual value of the control). So let's return all 3 of those back to what it was.

Reply:
i reverted changes with mask